packages/pyright-scip/snapshots/output/enclosing_range_class/advanced.py

```diff
@@ -0,0 +1,72 @@
+# < definition scip-python python snapshot-util 0.1 advanced/__init__:
+#documentation (module) advanced
+
+def class_decorator(cls):
+#   ^^^^^^^^^^^^^^^ definition <enclosing 0, 0, 3, 18> snapshot-util 0.1 advanced/class_decorator().
+#   documentation ```python
+#               > def class_decorator(
+#               >   cls
+#               > ): # -> (*args: Unknown, **kwargs: Unkno...
+#               > ```
+#                   ^^^ definition  snapshot-util 0.1 advanced/class_decorator().(cls)
+    def wrapper(*args, **kwargs):
+#       ^^^^^^^ definition <enclosing 1, 4, 2, 35> snapshot-util 0.1 advanced/class_decorator().wrapper().
+#       documentation ```python
+#                   > def wrapper(
+#                   >   *args,
+#                   >   **kwargs
+#                   > ):
+#                   > ```
+#                ^^^^ definition  snapshot-util 0.1 advanced/class_decorator().wrapper().(args)
+#                        ^^^^^^ definition  snapshot-util 0.1 advanced/class_decorator().wrapper().(kwargs)
+        return cls(*args, **kwargs)
+#              ^^^ reference  snapshot-util 0.1 advanced/class_decorator().(cls)
+#                   ^^^^ reference  snapshot-util 0.1 advanced/class_decorator().wrapper().(args)
+#                           ^^^^^^ reference  snapshot-util 0.1 advanced/class_decorator().wrapper().(kwargs)
+    return wrapper
+#          ^^^^^^^ reference  snapshot-util 0.1 advanced/class_decorator().wrapper().
```

Comment:
The point of this test case is to test the enclosing ranges.

1. The output around documentation is distracting from the result. In scip-clang, we support magic comments (https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/scip-clang%24%40main+format-options:&patternType=standard&sm=1&groupBy=path) to customize what is printed in the snapshot output. I would recommend adding support for that, so that we don't have doc comments distracting from what's actually under test.
2. If someone adds a test case in-between the existing test cases, then the numbers shown in the enclosing ranges for unrelated declarations will change. This is one of the reasons why the snapshots use `^^^` markers instead of actual numbers. I suggest changing the output style to use `^ start enclosing_range <symbol_name>` and `^ end enclosing_range <symbol_name>` instead. Cross-checking whether the line and character numbers are correct also requires looking at two files at once, the original and the snapshot file. The usage of carets would avoid that.

Reply:
Thanks for the feedback, @varungandhi-src 

Two things done:
- [ ] added support for magic comments in the form of `# format-options: showDocs, showRanges`
- [ ] switched to inline markets in the form of `⌄ start enclosing_range` and `⌃ end enclosing_range`.

Reply:
Thanks for the update, I'll take a deeper look in the next 24 hours.